controllers/operands/ssp.go

```diff
@@ -256,34 +257,50 @@ func getCommonDicts(list []hcov1beta1.DataImportCronTemplateStatus, crDicts map[
 		}
 
 		if crDict, found := crDicts[dictName]; found {
-			if !isDataImportCronTemplateEnabled(crDict) {
-				continue
-			}
-
 			// if the schedule is missing, copy from the common dict:
 			if len(crDict.Spec.Schedule) == 0 {
 				crDict.Spec.Schedule = targetDict.Spec.Schedule
 			}
 			targetDict.Spec = crDict.Spec.DeepCopy()
+			updateOrCreateMap(&targetDict.ObjectMeta.Annotations, crDict.Annotations)
+			updateOrCreateMap(&targetDict.ObjectMeta.Labels, crDict.Labels)
 			targetDict.ObjectMeta.Namespace = crDict.Namespace
 			targetDict.Status.Modified = true
 		} else if ns := hc.Spec.CommonBootImageNamespace; ns != nil && len(*ns) > 0 {
 			targetDict.ObjectMeta.Namespace = *ns
 		}
 
+		if !isDataImportCronTemplateEnabled(targetDict.DataImportCronTemplate) {
+			continue
+		}
 		list = append(list, targetDict)
 	}
 
 	return list
 }
 
+func updateOrCreateMap[M1 ~map[K]V, M2 ~map[K]V, K comparable, V any](originalPtr *M1, updates M2) {
```

Comment:
Why do we need any of this complexity when we know both callers just handle `map[string]string`?

Reply:
removed